### DISCUSSION TITLE
Is it a good idea to target latest opset version ?

### DISCUSSION THREAD

**Discussion by ghost** [onnx/onnx#5548] [Q&A] [answered]
Using `torch.onnx.export()`, and currently does run well in the browser with `onnxruntime`. 

Would it be missing performance improvements by using the default **v14** rather than the latest opset (**v20**) ?

I was especially curious because the code uses [Average Pool 2D](https://github.com/onnx/onnx/blob/main/docs/Operators.md), which has several releases (...,11, 19) 

Would it be advisable to use the latest instead ?

**Comment by cbourjau** [comment, marked as answer]
Newer versions often simply gain more attributes, convert attributes to inputs, or support more types. Generally speaking, runtimes will often share the same implementation of an operation between different versions. All that means is that you are unlikely to see performance gains. A much more important point is that runtimes may be lagging behind the latest opset version. The onnxruntime, for example, still only supports the `ai.onnx` domain up to (including) version 18.
